app/questionnaire/placeholder_parser.py

```diff
@@ -17,23 +20,23 @@ class PlaceholderParser:
 
     def __init__(
         self,
-        language,
-        schema=None,
-        answer_store=None,
-        metadata=None,
-        list_item_id=None,
-        location=None,
-        list_store=None,
+        language: str,
+        answer_store: AnswerStore,
+        list_store: ListStore,
+        metadata: ImmutableDict,
+        schema: QuestionnaireSchema,
```

Comment:
I didn't intentionally move schema, originally it had a default, which when doing the stores meant they had to move above (and put in the same order as value_source_resolver/ field_handlers), it wasn't till late that I realised that schema shouldn't have a default and could have stayed put, but by then I had already shifted it everywhere